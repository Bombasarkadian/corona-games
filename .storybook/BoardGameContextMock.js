import React from "react";
import { boolean, number, text, select } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { BoardGameProvider } from "contexts/BoardGameContext";

const ref = "Game Context";

const BoardGameContextMock = ({ children }) => {
  return (
    <BoardGameProvider
      G={{
        timePerTurn: 120,
        turnEndTime:
          Math.floor(new Date().getTime() / 1000) +
          number(
            "Remaining seconds",
            100,
            {
              range: true,
              min: 0,
              max: 120,
              step: 1,
            },
            ref
          ),
        points: [10, 3],
        actions: [
          { action: "message", text: text("Message text", "Hello there, shall we begin?", ref) },
        ],
        privateRoom: boolean("Private", true, ref),
        players: {
          "0": {
            phrase: text("Phrase", "Baba z wozu", ref),
            card: { pictures: [], layout: 0, rotation: 0 },
          },
          "1": {},
        },
        canChangePhrase: boolean("Can change phrase?", true, ref),
        currentCard: { pictures: [], layout: 0, rotation: 0 },
      }}
      ctx={{
        activePlayers: [
          select("stage", ["draw", "guess", "wait", "manage", "match"], "draw", ref),
          "guess",
        ],
        phase: select("phase", ["wait", "play"], "play", ref),
        numPlayers: number("Player count", 4, { range: true, min: 2, max: 10, step: 1 }, ref),
      }}
      moves={{
        Guess: action("Guess"),
        Ping: action("Ping"),
        UpdateDrawing: action("UpdateDrawing"),
        NotifyTimeout: action("NotifyTimeout"),
      }}
      playerID={0}
      gameName={"Kalambury"}
      gameID={"qwe123"}
      rawClient={{ transport: { socket: null } }}
      gameMetadata={[
        { id: 0, name: "user-0", isConnected: true },
        { id: 1, name: "user-1", isConnected: false },
      ]}
    >
      {children}
    </BoardGameProvider>
  );
};

export default BoardGameContextMock;
