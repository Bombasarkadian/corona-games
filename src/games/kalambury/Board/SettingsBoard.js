import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { Header, Form, Segment, Dropdown, Label } from "semantic-ui-react";

import WaitingBoard from "components/game/WaitingBoard";
import { phrasesSets } from "../data/phrases";
import moment from "moment";

const propTypes = {
  modes: PropTypes.arrayOf(PropTypes.string).isRequired,
  defaultMode: PropTypes.string,
  defaultTimePerTurn: PropTypes.number,
  onStartGame: PropTypes.func,
};

const SettingsBoard = ({ modes, defaultMode, defaultTimePerTurn, onStartGame }) => {
  const { t, i18n } = useTranslation("kalambury");

  const languages = phrasesSets.map((lang) => ({
    key: lang.key,
    value: lang.key,
    text: lang.name,
  }));
  const defaultLanguage = languages.find(({ value }) => value === i18n.language) || languages[0];

  const [gameMode, setGameMode] = useState(defaultMode);
  const [maxPoints, setMaxPoints] = useState(10);
  const [language, setLanguage] = useState(defaultLanguage.value);
  const [category, setCategory] = useState([]);
  const [timePerTurn, setTimePerTurn] = useState(defaultTimePerTurn);

  useEffect(() => {
    if (language) {
      setCategory([]);
    }
  }, [language, setCategory]);

  const categories = language
    ? phrasesSets
        .find(({ key }) => key === language)
        .categories.map((set) => ({
          key: set.key,
          value: set.key,
          text: set.name,
        }))
    : [];

  const handleStartGame = () => {
    onStartGame({
      gameMode,
      maxPoints,
      language,
      category,
      timePerTurn,
    });
  };

  const formatTime = (seconds) => {
    return moment.unix(seconds).format("m[min] ss[s]");
  };

  return (
    <WaitingBoard onStartGame={handleStartGame}>
      <Segment compact padded>
        <Form>
          <Header>{t("game.settings.mode")}</Header>
          {modes.map((mode) => (
            <Form.Field key={mode}>
              <Form.Radio
                toggle
                label={t(`game.settings.modes.${mode}`)}
                value={mode}
                checked={gameMode === mode}
                onChange={(e, { value }) => setGameMode(value)}
              />
            </Form.Field>
          ))}
          <Header disabled={gameMode === "infinite"} style={{ marginBottom: 5 }}>
            {t("game.settings.maxPoints")} <Label size="large">{maxPoints}</Label>
          </Header>
          <Form.Input
            fluid
            type="range"
            min="1"
            max="50"
            step="1"
            disabled={gameMode === "infinite"}
            value={maxPoints}
            onChange={(e, { value }) => setMaxPoints(value)}
          />
          <Header style={{ marginBottom: 5 }}>
            {t("game.settings.timePerTurn")} <Label size="large">{formatTime(timePerTurn)}</Label>
          </Header>
          <Form.Input
            fluid
            type="range"
            min="30"
            max="600"
            step="10"
            value={timePerTurn}
            onChange={(e, { value }) => setTimePerTurn(value)}
          />
          <Header>{t("game.settings.language")}</Header>
          <Form.Select
            options={languages}
            value={language}
            onChange={(e, { value }) => setLanguage(value)}
          />
          <Header>{t("game.settings.category")}</Header>
          <Dropdown
            selection
            multiple
            fluid
            placeholder="All categories"
            options={categories}
            value={category}
            onChange={(e, { value }) => setCategory(value)}
            disabled={categories.length === 0}
          />
        </Form>
      </Segment>
    </WaitingBoard>
  );
};
SettingsBoard.propTypes = propTypes;

export default React.memo(SettingsBoard);
