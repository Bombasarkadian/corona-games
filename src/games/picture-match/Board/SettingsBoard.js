import React, { useState } from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { Header, Image, Segment, Form } from "semantic-ui-react";

import WaitingBoard from "components/game/WaitingBoard";

const propTypes = {
  modes: PropTypes.arrayOf(PropTypes.string).isRequired,
  defaultMode: PropTypes.string,
  styles: PropTypes.arrayOf(PropTypes.string).isRequired,
  defaultStyle: PropTypes.string,
  defaultPicturesCount: PropTypes.number,
  onStartGame: PropTypes.func,
};

const SettingsBoard = ({
  modes,
  defaultMode,
  styles,
  defaultStyle,
  defaultPicturesCount,
  onStartGame,
}) => {
  const { t } = useTranslation("picture-match");

  const [gameMode, setGameMode] = useState(defaultMode);
  const [chosenStyle, setChosenStyle] = useState(defaultStyle);
  const [picturesCount, setPicturesCount] = useState(defaultPicturesCount);

  return (
    <WaitingBoard onStartGame={() => onStartGame(chosenStyle, gameMode, picturesCount)}>
      <Segment compact>
        <Form>
          <Header>{t("game.settings.style")}</Header>
          <Image.Group>
            {styles.map((style) => (
              <Image
                bordered
                key={style}
                style={{ cursor: "pointer", background: "white" }}
                src={`/images/games/picture-match/styles/${style}.png`}
                label={
                  chosenStyle === style
                    ? { as: "a", color: "green", corner: "left", icon: "check", size: "mini" }
                    : null
                }
                onClick={() => setChosenStyle(style)}
              />
            ))}
          </Image.Group>
          <Header>{t("game.settings.difficulty")}</Header>
          <Form.Field>
            <Form.Radio
              toggle
              label={t(`game.settings.difficulties.regular`)}
              value={8}
              checked={picturesCount === 8}
              onChange={(e, { value }) => setPicturesCount(8)}
            />
          </Form.Field>
          <Form.Field>
            <Form.Radio
              toggle
              label={t(`game.settings.difficulties.easy`)}
              value={6}
              checked={picturesCount === 6}
              onChange={(e, { value }) => setPicturesCount(6)}
            />
          </Form.Field>
          <Header>{t("game.settings.mode")}</Header>
          {modes.map((mode) => (
            <Form.Field key={mode}>
              <Form.Radio
                toggle
                label={t(`game.settings.modes.${mode}`)}
                value={mode}
                checked={gameMode === mode}
                onChange={(e, { value }) => setGameMode(value)}
              />
            </Form.Field>
          ))}
        </Form>
      </Segment>
    </WaitingBoard>
  );
};
SettingsBoard.propTypes = propTypes;

export default React.memo(SettingsBoard);
