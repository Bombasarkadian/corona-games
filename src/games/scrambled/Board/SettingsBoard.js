import React, { useState } from "react";
import PropTypes from "prop-types";
import { useTranslation } from "react-i18next";
import { Header, Segment, Form } from "semantic-ui-react";
import { availableLaguages } from "../data/tiles";

import WaitingBoard from "components/game/WaitingBoard";

const propTypes = {
  onStartGame: PropTypes.func,
};
const SettingsBoard = ({ onStartGame }) => {
  const { t, i18n } = useTranslation("scrambled");

  const languages = availableLaguages.map((lang) => ({
    key: lang,
    value: lang.key,
    text: lang.name,
  }));
  const defaultLanguage = languages.find(({ value }) => value === i18n.language) || languages[0];
  const [language, setLanguage] = useState(defaultLanguage.value);

  return (
    <WaitingBoard onStartGame={() => onStartGame(language)}>
      <Segment compact>
        <Form>
          <Header>{t("game.settings.language")}</Header>
          <Form.Select
            options={languages}
            value={language}
            onChange={(e, { value }) => setLanguage(value)}
          />
        </Form>
      </Segment>
    </WaitingBoard>
  );
};
SettingsBoard.propTypes = propTypes;

export default React.memo(SettingsBoard);
