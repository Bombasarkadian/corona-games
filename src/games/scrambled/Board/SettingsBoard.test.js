import React from "react";
import renderer from "react-test-renderer";
import SettingsBoard from "./SettingsBoard";
import BoardGameContextMock from "../GameContextMock";

describe("scrambled SettingsBoard component", () => {
  it("renders correctly", () => {
    expect.hasAssertions();
    let component;
    renderer.act(() => {
      component = renderer.create(
        <BoardGameContextMock overrideCtx={{ activePlayers: ["manage"] }}>
          <SettingsBoard
            assists={["assist1", "assist2"]}
            defaultAssist={"assist1"}
            StartGame={jest.fn()}
          />
        </BoardGameContextMock>
      );
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
