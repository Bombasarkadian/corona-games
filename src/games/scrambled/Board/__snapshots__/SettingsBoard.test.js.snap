// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrambled SettingsBoard component renders correctly 1`] = `
<div
  className="ui basic center aligned segment"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <div
    className="ui compact padded segment"
  >
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="ui header"
      >
        game.settings.language
      </div>
      <div
        className="field"
      >
        <div
          aria-describedby={null}
          aria-expanded={false}
          className="ui selection dropdown"
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="listbox"
          tabIndex={0}
        >
          <div
            aria-atomic={true}
            aria-live="polite"
            className="divider text"
            role="alert"
          >
            polski
          </div>
          <i
            aria-hidden="true"
            className="dropdown icon"
            onClick={[Function]}
          />
          <div
            className="menu transition"
          >
            <div
              aria-checked={true}
              aria-selected={true}
              className="active selected item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                polski
              </span>
            </div>
            <div
              aria-checked={false}
              aria-selected={false}
              className="item"
              onClick={[Function]}
              role="option"
              style={
                Object {
                  "pointerEvents": "all",
                }
              }
            >
              <span
                className="text"
              >
                English
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ui header"
      >
        game.settings.assist
      </div>
      <div
        className="field"
      >
        <div
          className="field"
        >
          <div
            className="ui checked toggle checkbox"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              aria-describedby={null}
              checked={true}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="radio"
              value="assist1"
            />
            <label>
              game.settings.assists.assist1
            </label>
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="field"
        >
          <div
            className="ui toggle checkbox"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              aria-describedby={null}
              checked={false}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="radio"
              value="assist2"
            />
            <label>
              game.settings.assists.assist2
            </label>
          </div>
        </div>
      </div>
      <div
        className="ui header"
      >
        game.settings.opponent_preview
        <div
          className="sub header"
          style={
            Object {
              "maxWidth": 260,
            }
          }
        >
          game.settings.opponent_preview_details
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="field"
        >
          <div
            className="ui checked toggle checkbox"
            onChange={[Function]}
            onClick={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
          >
            <input
              aria-describedby={null}
              checked={true}
              className="hidden"
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
            <label>
              game.settings.opponent_previews.true
            </label>
          </div>
        </div>
      </div>
    </form>
  </div>
  <button
    className="ui green icon right labeled button"
    onClick={[Function]}
  >
    board.wait.actions.start
    <i
      aria-hidden="true"
      className="pencil icon"
      onClick={[Function]}
    />
  </button>
</div>
`;
